File: stacks & queues/queue_using_stacks.py

class Queue:
    def __init__(self):
        self.head = []
        self.tail = []
    def enqueue(self, data):
        self.tail.append(data)
    def dequeue(self):
        if not self.head:
            while self.tail:
                self.head.append(self.tail.pop())
        return self.head.pop()
    def peek(self):
        if not self.head:
            while self.tail:
                self.head.append(self.tail.pop())
        return self.head[-1]

def main():
    q = Queue()
    q.enqueue(0)
    q.enqueue(1)
    print(q.peek())
    print(q.dequeue())
    print(q.peek())
    print(q.dequeue()) 
if __name__ == '__main__':
    main()

File: stacks & queues/queue_using_stacks.cpp
#include <iostream>
#include <stack>
using namespace std;

class MyQueue{
public:
  stack<int> stack1;
  stack<int> stack2;
  
  void enqueue(int x){
    while(!stack1.empty()){
      stack2.push(stack1.top());
      stack1.pop();
    }
    
    stack1.push(x);
    
    while(!stack2.empty()){
      stack1.push(stack2.top());
      stack2.pop();
    }
  }
  
  int dequeue() {
    if (stack1.empty()) {
      return -1; 
    }
    
    int removed = stack1.top();
    stack1.pop();
    
    return removed;
  }
  
  int peek() {
    if (stack1.empty()) {
      return -1; 
    }
    
    return stack1.top();
  }
  
};

// To execute C++, please define "int main()"
int main() {
  
  MyQueue mq;
  mq.enqueue(0);
  mq.enqueue(1);
  cout << mq.peek() << "\n";
  cout << mq.dequeue() << "\n";
  cout << mq.peek() << "\n";
  cout << mq.dequeue() << "\n";
  return 0;
}
  